# 대응 표본 T 검정 (Paired T-test)

표본의 각 사례마다 대응하는 2개의 관측치를 통해 판단

한 집단에 어떤 작용이 가해졌을 때에 대한 before,after를 비교

In [7]:
from scipy import stats
from pandas import read_excel

## 예제 (1) - 사슴의 다리 길이 비교

사슴의 왼쪽 뒷다리와 왼쪽 앞다리의 길이를 측정한 자료를 통해 두 다리의 길이(cm)가 같은지 다른지 알고 싶다.

이 경우 한 사슴에 대하여 두 다리 길이를 측정하였으므로 사슴은 독립이지만 한 마리의 사슴의 일부인 두 다리는 독립이 아니다.

In [8]:
df = read_excel("https://data.hossam.kr/E02/deer.xlsx", index_col="사슴")
df

뒷다리  앞다리
사슴          
1   142  138
2   140  136
3   144  147
4   144  139
5   142  143
6   146  141
7   149  143
8   150  145
9   142  136
10  148  146

Case 1

귀무가설 : 사슴의 앞다리와 뒷다리 길이에는 차이가 없다.

대립가설 : 사슴의 앞다리와 뒷다리 길이에는 차이가 있다.

In [9]:
stats.ttest_rel(df['앞다리'],df['뒷다리'],alternative='two-sided')

TtestResult(statistic=-3.413793103448276, pvalue=0.007703223347263766, df=9)

결과 해석
차이값에 대한 t 통계량의 값이 3.41, 대응되는 유의확률이 0.0077로 0.05보다 적으므로 유의수준 0.05에서 귀무가설을 기각, 앞다리와 뒷다리의 길이는 차이가 없지 않다고 결론 내릴 수 있다. 

Case 2 

귀무가설 : 사슴의 앞다리는 뒷다리보다 길지 않다

대립가설 : 사슴의 앞다리는 뒷다리보다 길다

In [10]:
stats.ttest_rel(df['앞다리'],df['뒷다리'],alternative='greater')

TtestResult(statistic=-3.413793103448276, pvalue=0.9961483883263681, df=9)

#### 결과 해석

차이값에 대한 t 통계량의 값이 `3.41`, 대응되는 유의확률이 `0.99`로 `0.05`보다 크므로 유의수준 `0.05`에서 앞다리는 뒷다리보다 길지 않다고 결론 내릴 수 있다.


Case 3 

귀무가설 : 사슴의 앞다리는 뒷다리보다 짧지 않다

대립가설 : 사슴의 앞다리는 뒷다리보다 짧다

In [11]:
stats.ttest_rel(df['앞다리'],df['뒷다리'],alternative='less')

TtestResult(statistic=-3.413793103448276, pvalue=0.003851611673631883, df=9)

#### 결과 해석

차이값에 대한 t 통계량의 값이 `3.41`, 대응되는 유의확률이 `0.0038`로 `0.05`보다 크므로 유의수준 `0.05`에서 앞다리는 뒷다리보다 짧다고 결론 내릴 수 있다.

## 예제 (2) - 자동차 첨가제가 주행거리에 미치는 영향

다섯 종류의 새 차에 대하여 같은 종류의 차 두 대 중에서 한 대를 랜덤하게 선택하여 첨가제를 사용하고, 다른 한 대에는 첨가제를 사용하지 않고 같은 운전자가 같은 장소에서 운전한 결과 다음과 같은 자료를 얻었다.

첨가제를 사용하면 주행거리가 늘어난다고 할 수 있는지 유의수준 5%에서 검정하여 보라.

> 그룹A : 첨가제를 사용한 경우, 그룹B : 첨가제를 사용하지 않은 경우

In [12]:
df = read_excel("https://data.hossam.kr/E02/oil.xlsx")
df

그룹A   그룹B
0  11.8  11.4
1  13.9  13.1
2  16.3  16.1
3  11.6  10.9
4   8.4   8.3

case 1 

귀무가설 : 그룹A와 그룹B의 주행거리에는 차이가 없다. 실험에 사용된 첨가제는 주행거리에 영향을 주지 않는다. 

대립가설 : 그룹A와 그룹B의 주행거리에는 차이가 있다. 실험에 사용된 첨가제는 주행거리에 영향을 준다.

In [13]:
stats.ttest_rel(df['그룹A'],df['그룹B'],alternative='two-sided')

TtestResult(statistic=3.226236530469768, pvalue=0.03208991550156059, df=4)

결과 해석

대응 표본 검정 결과에서 검정 통계량 값이 3.22이고 유의확률 p값이 0.032오 유의수준 0.05보다 작으므로 차이가 없다는 귀무가설을 기각한다. 따라서 첨가제를 사용하는 경우 주행거리에 차이가 있다고 판단 된다. 

case 2 

귀무가설 : 그룹A와 그룹B의 주행거리에는 차이가 없다. 실험에 사용된 첨가제는 주행거리 보다 짧지 않다.

대립가설 : 그룹A와 그룹B의 주행거리에는 차이가 있다. 실험에 사용된 첨가제는 주행거리 보다 짧다. 

In [14]:
stats.ttest_rel(df['그룹A'],df['그룹B'],alternative='less')

TtestResult(statistic=3.226236530469768, pvalue=0.9839550422492197, df=4)

결과 해석

대응 표본 검정 결과에서 검정 통계량 값이 3.22이고 유의확률 p값이 0.98로 유의수준 0.05보다 크므로 첨가제를 넣은 그룹의 주행거리가 첨가제를 넣지 않은 그룹의 구행거리보다 짧지 않다는 귀무가설을 기각할 수 없다.  

case 3 

귀무가설 : 그룹A와 그룹B의 주행거리에는 차이가 없다. 실험에 사용된 첨가제는 주행거리 보다 길지 않다. 즉, 첨가제는 주행거리를 늘리는데 효과가 없다. 

대립가설 : 그룹A와 그룹B의 주행거리에는 차이가 있다. 실험에 사용된 첨가제는 주행거리 보다 길다. 즉, 첨가제는 주행거리를 늘리는데 효과가 있다.  

In [15]:
stats.ttest_rel(df['그룹A'],df['그룹B'],alternative='greater')

TtestResult(statistic=3.226236530469768, pvalue=0.016044957750780294, df=4)

결과 해석

대응 표본 검정 결과에서 검정 통계량 값이 3.22이고 유의확률 p값이 0.01로 유의수준 0.05보다 작므로 첨가제를 넣은 그룹의 주행거리가 첨가제를 넣지 않은 그룹의 구행거리보다 길지 않다는 귀무가설을 기각할 수 있고 대립가설을 채택할 수 있다. 즉, 첨가제는 주행거리 연장에 효과가 있다. 